lib/ansible/modules/net_tools/ipify_facts.py

```diff
@@ -92,19 +93,20 @@ def run(self):
         result = {
             'ipify_public_ip': None
         }
-        (response, info) = fetch_url(module=module, url=self.api_url + "?format=json" , force=True, timeout=self.timeout)
+        (response, info) = fetch_url(module=module, url=self.api_url + "?format=json", force=True, timeout=self.timeout)
 
         if not response:
             module.fail_json(msg="No valid or no response from url %s within %s seconds (timeout)" % (self.api_url, self.timeout))
 
-        data = json.loads(response.read())
+        data = json.loads(to_text(response.read()))
         result['ipify_public_ip'] = data.get('ip')
         return result
 
+
 def main():
     global module
     module = AnsibleModule(
-        argument_spec = dict(
+        argument_spec=dict(
             api_url=dict(default='https://api.ipify.org'),
             timeout=dict(type='int', default=10),
```

Comment:
Great, the only thing to do is to add the param `timeout=dict(type='bool', default=True),` (and the docs for the new param)